src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.TransactionsRecording;
+using Raven.Client.Documents.Session;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class TransactionsRecordingHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/transactions/replay", "POST", AuthorizationStatus.ValidUser)]
+        public async Task ReplayRecording()
+        {
+            var operationId = GetLongQueryString("operationId");
```

Comment:
The operation should not be passed from the outside (unless used via Studio).

This endpoint should return the OperationId only (check other similar methods e.g. in SmugglerHandler)

The ReplayOperation in client should return OperationIdResult

Reply:
As I understand it, the request is executed while the file stream is uploaded.
So, if the OperationId will be passed at the end of the request the client will not be able to monitor the operation that can take a lot of time.